.github/workflows/ci.yml

```diff
@@ -77,40 +79,28 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
       with:
-        fetch-depth: 0
-
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python }}
-        cache: pip
+        persist-credentials: false
 
     - name: Install Graphviz for doc builds
-      if: startsWith(matrix.name, 'Documentation')
+      if: contains(matrix.nox_session, 'docs')
       uses: ts-graphviz/setup-graphviz@v2
 
-    - name: Install pandoc for doc builds (requires Ubuntu)
-      if: startsWith(matrix.name, 'Documentation')
+    - name: Install pandoc for doc builds
+      # This step requires Ubuntu
+      if: contains(matrix.nox_session, 'docs')
       run: |
         sudo apt update
         sudo apt-get install -y pandoc
 
-    - name: Install Nox and uv
-      run: python -m pip install --progress-bar off --upgrade nox uv
-
-    - name: Cache .nox to shorten virtual environment creation time
-      uses: actions/cache@v4
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
       with:
-        path: |
-          .nox
-          .venv
-          .cache/uv
-          cache/uv
-        # Changing pyproject.toml or requirements invalidates the cache
-        key: ${{ matrix.nox_session }}-${{ matrix.python }}-${{ runner.os }}-${{ hashFiles('pyproject.toml', 'uv.lock') }}
-
-    - name: Run Nox session
-      run: nox -s '${{ matrix.nox_session }}'
+        python-version: ${{ matrix.python }}
+        enable-cache: true
+        cache-suffix: ${{ matrix.nox_session }}-${{ matrix.python }}-${{ runner.os }}
+
+    - name: Run the check
+      run: uv run --python ${{ matrix.python }} --with nox --frozen nox -s '${{ matrix.nox_session }}'
```

Comment:
Do you need `--python ${{ matrix.python }}` since `setup-uv` already declares the python version to use?

Reply:
Looking at the docs for [`setup-uv`](https://github.com/astral-sh/setup-uv?tab=readme-ov-file#python-version), it appears that we don't need to use `--python` if we already declare it using `with:` in `setup-uv`. I'll revise this.  Thanks!